### PR TITLE
Fixes #25973: kafkaconnect add container entity wildcard search

### LINKED ISSUE #25973: Kafkaconnect lineage should also consider file names for storage services
**Affected module**
Does it impact the UI, backend or Ingestion Framework?

**Describe the bug**

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by SumanMaharana**
### Describe your changes:
Fixes https://github.com/open-metadata/OpenMetadata/issues/25973
This pull request improves the robustness of container entity lookup in the Kafka Connect ingestion source, specifically addressing cases where container names may include topic name suffixes. It introduces a fallback wildcard search mechanism in the entity resolution logic and adds comprehensive unit tests to ensure correct behavior in various scenarios.

**Enhancements to container entity lookup:**

* Added a fallback wildcard search in `get_dataset_entity` to handle cases where containers are ingested with topic name suffixes, improving entity resolution when exact FQN matches fail.

**Testing improvements:**

* Introduced a new test class `TestGetDatasetEntityContainerSearch` with multiple test cases covering both exact and wildcard search paths, including edge cases such as missing container names, correct pattern construction, and scoping searches to specific storage services.

#
### Type of change:

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

----
## Summary by Gitar

- **Fixed container entity lookup:** Added fallback wildcard search in `get_dataset_entity()` to handle containers ingested with topic name suffixes (e.g., `raw_kafka/topic-01.v1`), resolving FQN mismatches in Kafka Connect S3 sink scenarios
- **Comprehensive test coverage:** Introduced `TestGetDatasetEntityContainerSearch` class with 9 test cases covering exact match paths, wildcard fallback triggering, parameter scoping, and edge cases (missing container names, parent containers)

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/pipeline/kafkaconnect/metadata.py [hunk 1]
```diff
@@ -620,6 +620,35 @@ def get_dataset_entity(
...
+                    # This handles cases where containers are ingested with topic names appended
+                    # Example: StorageServiceName.BucketName.Directory/TopicName.extension
+                    if storage_entity is None and dataset_details.container_name:
+                        logger.debug(
+                            f"Exact container match failed, trying prefix search for "
+                            f"container '{dataset_details.container_name}*' "
+                            f"under parent '{dataset_details.parent_container}'"
+                        )
+
+                        for storageservicename in self.get_storage_service_names() or [
```
[the remark this answers is not in the document]
Comment (SumanMaharana) [reply ingestion/src/metadata/ingestion/source/pipeline/kafkaconnect/metadata.py:633]: `fqn.search_container_from_es` accepts `None` here not `*`
